twilight-util/src/signature_validation.rs

```diff
@@ -0,0 +1,184 @@
+//! Provides signature validation as is required for bots which work by giving Discord
+//! an HTTPS endpoint to send Interactions to.
+//!
+//! See <https://discord.com/developers/docs/interactions/receiving-and-responding#security-and-authorization>
+//! for more details.
+
+use ed25519_dalek::{Signature, SignatureError, VerifyingKey};
+#[cfg(feature = "signature-validation-extract-interaction")]
+use twilight_model::application::interaction::Interaction;
+
+/// Parsing a hexadecimal string failed.
+#[derive(Debug)]
+pub struct FromHexError(hex::FromHexError);
```

Comment:
Let's specialize this for `Key::from_hex`
```suggestion
pub struct KeyParseError(hex::FromHexError);
```